source/panel_predictor/panel_predictor.py

```diff
@@ -0,0 +1,47 @@
+from source.common.module import Module
+from pathlib import Path
+import os
+from source.panel_predictor.panel_finder.panel_finder import PanelFinder
+import time
+
+
+class PanelPredictor(Module):
+    def __init__(self, parent=None, state=None):
+        self.working_dir = Path(os.path.dirname(os.path.abspath(__file__)))
+        super().__init__(self.working_dir, parent=parent, state=state)
+        self.panel = None
+        self.panel_finder = PanelFinder(self, state=state)  # this panel finder needs no additional properties
+        self.past_targets = []
+
+    def process(self, frame):
+        panel = self.panel_finder.process(frame)
+        if panel is not None:
+            panel, confidence = panel
+            if len(self.past_targets) == self.properties["reference_frames"]:
+                self.past_targets.pop(0)
+            self.past_targets.append((panel["x_center"], panel["y_center"], time.time(), confidence))
+        if len(self.past_targets) == self.properties["reference_frames"]:
+            if self.properties["prediction_type"] == "linear":
+                velocity, distance_confidence = self.average_velocity(((frame.shape[0]) ** 2 + (frame.shape[1]) ** 2) ** (1 / 2))
```

Comment:
I think this linear prediction code should be put in a separate function. It's easier to understand when separated.